sympy/physics/control/tests/test_lti.py

```diff
@@ -577,53 +574,186 @@ def test_Feedback_construction():
-    assert (tf1*tf2) / (tf + tf1*tf2*tf5*tf3) in (Feedback(tf1*tf2, tf5*tf3), Feedback(tf1*tf2, tf3*tf5))
+    assert tf / (tf + TF1) == Feedback(tf, TF1)
+    assert tf / (tf + TF1*TF2*TF3) == Feedback(tf, TF1*TF2*TF3)
+    assert TF1 / (tf + TF1*TF2*TF3) == Feedback(TF1, TF2*TF3)
+    assert (TF1*TF2) / (tf + TF1*TF2) == Feedback(TF1*TF2, tf)
+    assert (TF1*TF2) / (tf + TF1*TF2*tf5) == Feedback(TF1*TF2, tf5)
+    assert (TF1*TF2) / (tf + TF1*TF2*tf5*TF3) in (Feedback(TF1*TF2, tf5*TF3), Feedback(TF1*TF2, TF3*tf5))
     assert tf4 / (TransferFunction(1, 1, p) + tf4*tf6) == Feedback(tf4, tf6)
     assert tf5 / (tf + tf5) == Feedback(tf5, tf)
 
-    raises(ValueError, lambda: tf1*tf2*tf3 / (1 + tf1*tf2*tf3))
-    raises(ValueError, lambda: tf1*tf2*tf3 / tf3*tf5)
-    raises(ValueError, lambda: tf2*tf3 / (tf + tf2*tf3*tf4))
+    raises(ValueError, lambda: TF1*TF2*TF3 / (1 + TF1*TF2*TF3))
+    raises(ValueError, lambda: TF1*TF2*TF3 / TF3*tf5)
+    raises(ValueError, lambda: TF2*TF3 / (tf + TF2*TF3*tf4))
 
-    assert Feedback(tf, tf1*tf2*tf3).doit() == \
+    assert Feedback(tf, TF1*TF2*TF3).doit() == \
         TransferFunction((a2*s + p)*(s**2 + 2*s*wn*zeta + wn**2), k*(a2*p - s) + \
         (a2*s + p)*(s**2 + 2*s*wn*zeta + wn**2), s)
-    assert Feedback(tf1, tf2*tf3).doit() == \
+    assert Feedback(TF1, TF2*TF3).doit() == \
         TransferFunction((a2*s + p)*(s**2 + 2*s*wn*zeta + wn**2), (k*(a2*p - s) + \
         (a2*s + p)*(s**2 + 2*s*wn*zeta + wn**2))*(s**2 + 2*s*wn*zeta + wn**2), s)
-    assert Feedback(tf1*tf2, tf5).doit() == \
+    assert Feedback(TF1*TF2, tf5).doit() == \
         TransferFunction(k*(a0 + s)*(s**2 + 2*s*wn*zeta + wn**2), (k*(-a0 + a1*s**2 + a2*s) + \
         (a0 + s)*(s**2 + 2*s*wn*zeta + wn**2))*(s**2 + 2*s*wn*zeta + wn**2), s)
     assert Feedback(tf4, tf6).doit() == \
         TransferFunction(p*(p + s)*(a0*p + p**a1 - s), p*(p*(p + s) + (-p + s)*(a0*p + p**a1 - s)), p)
     assert -Feedback(tf4*tf6, TransferFunction(1, 1, p)).doit() == \
         TransferFunction(-p*(-p + s)*(p + s)*(a0*p + p**a1 - s), p*(p + s)*(p*(p + s) + (-p + s)*(a0*p + p**a1 - s)), p)
 
-    assert Feedback(tf1, tf2*tf5).rewrite(TransferFunction) == \
+    assert Feedback(TF1, TF2*tf5).rewrite(TransferFunction) == \
         TransferFunction((a0 + s)*(s**2 + 2*s*wn*zeta + wn**2), (k*(-a0 + a1*s**2 + a2*s) + \
         (a0 + s)*(s**2 + 2*s*wn*zeta + wn**2))*(s**2 + 2*s*wn*zeta + wn**2), s)
     assert Feedback(TransferFunction(1, 1, p), tf4).rewrite(TransferFunction) == \
         TransferFunction(p, a0*p + p + p**a1 - s, p)
+
+
+def test_TransferFunctionMatrix_construction():
+    tf5 = TransferFunction(a1*s**2 + a2*s - a0, s + a0, s)
+    tf4 = TransferFunction(a0*p + p**a1 - s, p, p)
+    tf6 = TransferFunction(a0*s**3 + a1*s**2 - a2*s, b0*p**4 + b1*p**3 - b2*s*p, s)
+    tf7 = TransferFunction(p, a0, p)
+
+    tfm1 = TransferFunctionMatrix([TF1, TF2], (2, 1), s)
+    assert tfm1.shape == (tfm1.outputs, tfm1.inputs) == (2, 1)
```

Comment:
The attribute names "outputs" and "inputs" imply that they'd return what they are. But this returns the number of outputs and number of inputs. I think that "num_outputs" and "num_inputs" are better name choices (or something similar.